### PR TITLE
Give prevalence to process level over primary/module levels

### PR DESCRIPTION
**Pull request by v0idpwn**
The docs incorrectly implied that `Logger.put_process_level/2` allowed one to decrease the logger level for a process. This is not possible with this function.

### CODE REVIEW

**Comment by v0idpwn** [comment]
Example:

```
Erlang/OTP 27 [erts-15.0] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit:ns]

Interactive Elixir (1.17.1) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> require Logger
Logger
iex(2)> Logger.configure(level: :info)
:ok
iex(3)> Logger.put_process_level(self(), :debug)
:ok
iex(4)> Logger.debug("hello?")
:ok
iex(5)> Logger.info("info")

09:46:25.063 [info] info
:ok
iex(6)> Logger.put_process_level(self(), :error)
:ok
iex(7)> Logger.info("info again")
:ok
iex(8)> Logger.error("error")

09:46:47.350 [error] error
:ok
iex(9)>
```

**Comment by josevalim** [comment]
It seems the `put_module_level` makes a bypass, so if that's the case, maybe we should try to mirror it here?

**Comment by v0idpwn** [comment]
Pushed the change to prefer process level over global level. Will write a test.

File: lib/logger/lib/logger.ex [hunk 1]
```diff
@@ -939,8 +939,14 @@ defmodule Logger do
   def __should_log__(level, module) do
     level = elixir_level_to_erlang_level(level)
 
-    if :logger_config.allow(level, module) do
-      level
+    case get_process_level(self()) do
```
Comment (josevalim) [inline lib/logger/lib/logger.ex:942]: We need to apply the same to  `bare_log` after the recent commits and add tests :)

File: lib/logger/lib/logger.ex [hunk 2]
```diff
@@ -939,8 +939,14 @@ defmodule Logger do
...
+      nil ->
+        if :logger_config.allow(level, module) do
+          level
+        end
+
+      process_level ->
```
Comment (josevalim) [inline lib/logger/lib/logger.ex:948]: We still need to check if this level is allowed by either calling `:logger_config.allow(level)` or by calling `:logger_config.allow(level, module)` if we want still want to look up the module.

File: lib/logger/test/logger_test.exs [hunk 3]
```diff
@@ -760,7 +784,7 @@ defmodule LoggerTest do
 
     test "maps Erlang levels" do
       :logger.set_primary_config(:level, :notice)
-      assert capture_log(fn -> Logger.info("hello") end) =~ "hello"
+      assert capture_log(fn -> Logger.warning("hello") end) =~ "hello"
```
Comment (v0idpwn) [inline lib/logger/test/logger_test.exs:787]: I'm not sure what the behaviour was being tested here. This started failing when I made `capture_log` not change log level

File: lib/logger/lib/logger.ex (hunk 2 above)
Comment (v0idpwn) [reply lib/logger/lib/logger.ex:955]: Calling :logger_config.allow(level, module) there would give precedence for module/primary level. Pushed some fixes and I believe we have the expected behaviour now :)

**Comment by v0idpwn** [comment]
There's a big problem with this approach: elixir and erlang logs behave under different rules. I think this is not desirable and probably we can't merge.

**Pull request closed without merging**